options/configurable.cc

```diff
@@ -0,0 +1,606 @@
+    const std::unordered_map<std::string, std::string>& opts_map,
+    std::unordered_map<std::string, std::string>* unused) {
+  std::unordered_map<std::string, std::string> remaining = opts_map;
+  Status s = Status::OK();
+  if (!opts_map.empty()) {
+#ifndef ROCKSDB_LITE
+    for (const auto& iter : configurable.options_) {
+      s = ConfigureSomeOptions(config_options, configurable, *(iter.type_map),
+                               &remaining, iter.opt_ptr);
+      if (remaining.empty()) {  // Are there more options left?
+        break;
+      } else if (!s.ok()) {
+        break;
+      }
+    }
+#else
+    (void)configurable;
+    if (!config_options.ignore_unknown_options) {
+      s = Status::NotSupported("ConfigureFromMap not supported in LITE mode");
+    }
+#endif  // ROCKSDB_LITE
+  }
+  if (unused != nullptr && !remaining.empty()) {
+    unused->insert(remaining.begin(), remaining.end());
+  }
+  if (config_options.ignore_unknown_options) {
+    s = Status::OK();
+  } else if (s.ok() && unused == nullptr && !remaining.empty()) {
+    s = Status::NotFound("Could not find option: ", remaining.begin()->first);
+  }
+  return s;
+}
+
+#ifndef ROCKSDB_LITE
+/**
+ * Updates the object with the named-value property values, returning OK on
+ * succcess. Any properties that were found are removed from the options list;
+ * upon return only options that were not found in this opt_map remain.
+ * Returns OK if the properties were successfully updated, N
+ */
+Status ConfigurableHelper::ConfigureSomeOptions(
+    const ConfigOptions& config_options, Configurable& configurable,
+    const std::unordered_map<std::string, OptionTypeInfo>& type_map,
+    std::unordered_map<std::string, std::string>* options, void* opt_ptr) {
+  Status result = Status::OK();
+  Status notsup = Status::OK();
+  std::string elem_name;
+  int found = 1;
+  std::unordered_set<std::string> unsupported;
+  // While there are unused properties and we processed at least one,
+  // go through the remaining unused properties and attempt to configure them.
+  while (found > 0 && !options->empty()) {
+    found = 0;
+    notsup = Status::OK();
+    for (auto it = options->begin(); it != options->end();) {
+      const std::string& opt_name = configurable.GetOptionName(it->first);
+      const std::string& opt_value = it->second;
+      const auto opt_info =
+          OptionTypeInfo::Find(opt_name, type_map, &elem_name);
+      if (opt_info == nullptr) {  // Did not find the option.  Skip it
+        ++it;
+      } else {
+        Status s = ConfigureOption(config_options, configurable, *opt_info,
+                                   opt_name, elem_name, opt_value, opt_ptr);
+        if (s.IsNotFound()) {
+          ++it;
+        } else if (s.IsNotSupported()) {
+          notsup = s;
+          unsupported.insert(it->first);
+          ++it;  // Skip it for now
+        } else {
+          found++;
+          it = options->erase(it);
+          if (!s.ok()) {
```

Comment:
In this case, `result` stores the last non-ok status?

Reply:
Yes, added a comment on their declaration